### ISSUE TITLE
java.lang.AssertionError: Already had dead JAR

### ISSUE THREAD

**Issue by bruceleecn** [sleuthkit/autopsy#2873]
Hello,
 I trying to start autopsy 4.4.0 on **Windows 7**. I've done all steps from documentation and I've seen the liitle window with dog and application starting progress bar, but I've got this stacktrace insted application UI.

`java.lang.AssertionError: Already had dead JAR: C:\Users\user\AppData\Local\Temp\org-netbeans-api-progress5910661202203464386.jar
	at org.netbeans.JarClassLoader$JarSource.destroy(JarClassLoader.java:700)
	at org.netbeans.JarClassLoader.destroy(JarClassLoader.java:341)
	at org.netbeans.StandardModule.classLoaderDown(StandardModule.java:581)
	at org.netbeans.ModuleManager.enable(ModuleManager.java:1314)
	at org.netbeans.ModuleManager.enable(ModuleManager.java:1170)
	at org.netbeans.core.startup.ModuleList.installNew(ModuleList.java:340)
	at org.netbeans.core.startup.ModuleList.trigger(ModuleList.java:276)
	at org.netbeans.core.startup.ModuleSystem.restore(ModuleSystem.java:301)
	at org.netbeans.core.startup.Main.getModuleSystem(Main.java:181)
	at org.netbeans.core.startup.Main.getModuleSystem(Main.java:150)
	at org.netbeans.core.startup.Main.start(Main.java:307)
	at org.netbeans.core.startup.TopThreadGroup.run(TopThreadGroup.java:123)
	at java.lang.Thread.run(Thread.java:745)

Result: 2`

**Comment by rcordovano** [comment]
@bruceleecn, did you build from source or are you using a Windows installer?

Richard Cordovano
Autopsy Team Lead
Director of Engineering - Digital Forensics, Basis Technology

**Comment by bruceleecn** [comment]
@rcordovano Thanks for your help.
I have solved it. The cause of the error was that I imported the wrong file "libtsk_jni.dll". after I rebuild it with VS, the error is gone, and autopsy can be started.

Thanks a lot.

**Comment by rcordovano** [comment]
@bruceleecn, you're welcome, although you solved it yourself.

For future reference, I was asking about whether you were building from source because this "already had dead JAR" message comes from NetBeans clean up code that runs when module loading fails at run time. My team has experienced a wide variety of errors for which this exception is the last word, but provides no help in debugging. Here are a few of the error conditions:

- Mixing of debug and release builds of the SleuthKit and Autopsy so that library files are not found
- Incorrect Installer class path in manifest for an NBM in the Autopsy modules suite
- Attempting to start the application in the IDE with an installed version running (for this one, the copy of the JAR referenced in the exception was already written by the running instance, so this error condition is actually directly related)
- Removal of JARs from the list of disabled modules in platform.properties

Sincerely,
Richard Cordovano
Autopsy Team Lead
Director of Engineering - Digital Forensics, Basis Technology

**Comment by xkill** [comment]
I have the same problem from clean repositories:
```bash
cd /usr/local/src/
git clone git://github.com/sleuthkit/sleuthkit.git
cd sleuthkit
git checkout release-4.4.2
./bootstrap
./configure
make
sudo make install
cd bindings/java
ant dist-PostgreSQL
cp /usr/local/src/sleuthkit/bindings/java/dist/Tsk_DataModel.jar /usr/local/src/sleuthkit/bindings/java/dist/Tsk_DataModel_PostgreSQL.jar   #to a void a future problem, cos this library name is changed.
```
```
cd ..
export TSK_HOME= /usr/local/src/sleuthkit
git clone git@github.com:sleuthkit/autopsy.git
cd autopsy
git checkout release-4.4.1
ant
ant run
[...]
     [exec] WARNING [org.netbeans.core.startup.NbEvents]: The extension /usr/local/src/autopsy/build/cluster/modules/ext/commons-logging-1.2.jar may be multiply loaded by modules: [/usr/local/src/autopsy/build/cluster/modules/org-sleuthkit-autopsy-centralrepository.jar, /usr/local/src/autopsy/build/cluster/modules/org-sleuthkit-autopsy-keywordsearch.jar]; see: http://www.netbeans.org/download/dev/javadoc/org-openide-modules/org/openide/modules/doc-files/classpath.html#class-path
     [exec] Temp Folder for Libraries: /tmp
     [exec] java.lang.AssertionError: Already had dead JAR: /tmp/org-netbeans-api-progress3561386651767449064.jar
     [exec] 	at org.netbeans.JarClassLoader$JarSource.destroy(JarClassLoader.java:700)
     [exec] 	at org.netbeans.JarClassLoader.destroy(JarClassLoader.java:341)
     [exec] 	at org.netbeans.StandardModule.classLoaderDown(StandardModule.java:581)
     [exec] 	at org.netbeans.ModuleManager.enable(ModuleManager.java:1314)
     [exec] 	at org.netbeans.ModuleManager.enable(ModuleManager.java:1170)
     [exec] 	at org.netbeans.core.startup.ModuleList.installNew(ModuleList.java:340)
     [exec] 	at org.netbeans.core.startup.ModuleList.trigger(ModuleList.java:276)
     [exec] 	at org.netbeans.core.startup.ModuleSystem.restore(ModuleSystem.java:301)
     [exec] 	at org.netbeans.core.startup.Main.getModuleSystem(Main.java:181)
     [exec] 	at org.netbeans.core.startup.Main.getModuleSystem(Main.java:150)
     [exec] 	at org.netbeans.core.startup.Main.start(Main.java:307)
     [exec] 	at org.netbeans.core.startup.TopThreadGroup.run(TopThreadGroup.java:123)
     [exec] 	at java.lang.Thread.run(Thread.java:748)
     [exec] Result: 2

BUILD SUCCESSFUL
Total time: 1 minute 29 seconds
```

I tried to do the same with netbeans, instead ant, but same error.

**Comment by jmvbxx** [comment]
I wouldn't hold your breath @xkill for a response. For reasons that confound me, this project does not support non-Windows users.

**Comment by bcarrier** [comment]
@jmvbxx, as I mentioned in #2819, the "project" would love to have Autopsy working on all platforms, but the Basis Tech team is currently focused on the Windows platform.  We try to provide guidance whenever we can to those who have issues on Linux and OS X, but 99% of our time is spent on Windows because that is where our customers are paying us to focus on.  

If someone in the community has the time and interest to debug this specific issue, we'd certainly welcome the pull request for the fixes.

**Comment by jmvbxx** [comment]
@bcarrier I respect that every project/organization has limited resources which is why many choose to offer community editions of their software. I recognize, and appreciate, that this is something that you've chosen to do.

However, as a Linux user I find that not even the smallest effort is made to accommodate my operating system with the Linux "version" being two full versions behind or even a lack of basic documentation for non-Windows users. (Yes, there's build docs in the repo).

From your answer I'm going to guess that you are a Windows user and have not experienced the community feeling outside of closed software. I'm sure if you made even the smallest attempts (not even in software, but in nicer documentation, a FAQ, anything, you'd find the community would step in and pick up a lot of the slack.

**Comment by mckeowbc** [comment]
@bcarrier it looks like whatever was causing the "dead jar" issue has been resolved in Autopsy 4.5.0.  My friend Tim and I both got it working at #OSDFCon on Ubuntu 16.04.

https://oylenshpeegul.wordpress.com/2017/10/18/autopsy-4-5-works-in-linux/
